docs/CONTRIBUTING.md

```diff
@@ -102,8 +121,22 @@ In addition to the guidelines covered in the Development Environment Setup Instr
-6. Commit messages have to be related to a specific issue on GitHub and reference its issue number as well as include a short description on what has been done in the commit. We might reject pull requests that violate this principle and ask you to re-submit it with proper commit messages. An acceptable commit message could look like this:
-   > #61 - added model to define classification, apply automatic case classification whenever a field value changes
+
+#### Rules for commits
+
+1. Commit messages of **every commit** have to be related to a specific issue on GitHub and reference its issue number as well as include a short description on what has been done in the commit. We will reject pull requests that violate this principle and ask you to re-submit it with proper commit messages. An acceptable commit message could look like this:
+   > #1234 Added model to define classification
+2. A common practice for Git commits is to keep the first line with 50 characters as meaninful short description, and to deliver additional details following in line 3 onwards. Most git viewers abbreviate the first line after 50 characters.
+   > #1234 Added model to define classification
+   >
+   > - Apply automatic case classification whenever a field value changes
+   > - Show classification in lists
+3. Keep changes in bugfixes clean and compact to be able to easily review them and leave the possibility to cherry-pick the changes to another branch to fix a previous version. Don't clean dirty code along the way if not really needed to fix the problem.
+4. If an issue requires a lot of code changes, consider breaking down these changes in logical steps. They are then easier to review, have more meaningful commit messages and deliver partly the intended value.
+5. Don't mix refactoring with functional changes (new functionality, changes on features, bugfixes) within the same commit, since it makes reviewing the changes much harder. Usually the refactoring of existing code has to happen beforehand in at least one separate commits. This means refactoring (no functional change): Cleaner code, renaming, restructuring.
+6. If it helps, it is okay to have several branches and pull requests for the same ticket (usually one after another, sometimes to work in parallel or to prepare changes in advance).
+7. If you feel an issue consists out of several parts, feel free to document this in the **Implementation Details** as task list and check what you consider as finished. Do not break down an issue into sub-issues, this is to clarified with the development team.
+8. If there is a finding concerning your already closed issue, it will be reopened if that version is not yet released and you or someone else will have to quickly fix the finding before the version is released. If the version was already released, the issue will be kept closed and a new issue has to be documented.
```

Comment:
concerning an assign ticket which has been already closed, it will be reopened if that version is not yet released.  You (or someone else) will have to quickly fix the finding before the version is released. If the version has been released already, the issue will stay closed and a new issue has to be filed.